frontend/editor/index.html

```diff
@@ -7,78 +7,104 @@
-	</select>
-	<br>
-	<label for="period_class_index">Class Index</label>
-	<input id="period_class_index" type="number">
-	<script src="../editor/controls-section.js"></script>
+	<div id="settings">
+		<button id="pull">Fetch Schedules</button>
+		<button id="push">Apply Changes</button>
+		<button id="update">Force Schedule Reload</button>
+		<span id="controls_error" class="error"></span>
+		<hr>
+		<button id="export">Export settings</button>
+		<br>
+		<label for="import">Import settings:</label>
+		<input type="file" id="import" accept=".json" />
+		<hr>
+		<h2>Manage Schedules</h2>
+		<button id="add_schedule">Create New Blank Schedule</button>
+		<br>
+		<select id="select_schedule">
+		</select>
+		<button id="copy_schedule">Duplicate Selected Schedule</button>
+		<button id="remove_schedule">Delete Selected Schedule</button>
+		<br>
+		<br>
+		<label for="schedule_friendly_name">Schedule Name</label>
+		<input id="schedule_friendly_name" class="schedule_setting">
+		<br>
+		<label for="schedule_code">Schedule Code (internal use)</label>
+		<input id="schedule_code" class="schedule_setting">
+		<br>
+		<label for="schedule_color">Schedule Color</label>
+		<input id="schedule_color" class="schedule_setting" type="color">
+		<br>
+		<label for="schedule_hide">Hide in dropdowns</label>
+		<input id="schedule_hide" class="schedule_setting" type="checkbox">
+		<br>
+		<label for="schedule_regex">Schedule Regex</label>
+		<input id="schedule_regex" class="schedule_setting">
+		<br>
+		<hr>
+		<h2>Schedule Calendar</h2>
+		<label for="calendars">Calendar URLs (ical feed, one per line. Top items take priority.)</label>
+		<br>
+		<textarea id="calendars"></textarea>
+		<br>
+		<h3>Typical Week</h3>
+		<div id="typical_schedule">
+
+		</div>
+		<hr>
+		<h2>Manage Periods</h2>
+		<template id="period_settings">
+			<button id="move_up">Move Up</button>
+			<button id="move_down">Move Down</button>
+			<button id="remove_period">Remove</button>
+			<br>
+			<label for="period_friendly_name">Friendly Name</label>
+			<input id="period_friendly_name">
+			<br>
+			<label for="period_start">Start Time</label>
+			<input type="time" id="period_start">
+			<br>
+			<label for="period_end">End Time</label>
+			<input type="time" id="period_end">
+			<br>
+			<label for="period_kind">Period Kind</label>
+			<select id="period_kind">
+				<option value="Class">Class</option>
+				<option value="Lunch">Lunch</option>
+				<option value="Break">Break</option>
+				<option value="AMSupport">AM Support</option>
+				<option value="Passing">Passing (implicit)</option>
+				<option value="BeforeSchool">BeforeSchool (implicit)</option>
+				<option value="AfterSchool">AfterSchool (implicit)</option>
+			</select>
+			<br>
+			<label for="period_class_index">Class Code (internal use)</label>
+			<input id="period_class_index">
+		</template>
+		<div id="periods">
+		</div>
+		<button id="add_period" disabled>Add Period</button>
```

Comment:
This button looks pretty similar to the things in the periods above it so it ends up being somewhat invisible (not literally) in the layout, but I don't have any ideas for a better place for it.

Reply:
I could add a `hr` to break it up a bit but I don't know if that would help

Reply:
```suggestion
		<button id="add_period" disabled style="margin-top: 5px; padding: 6px 10px;">Add Period</button>
```
I think it would help if we made the button bigger and added a bit more space between it and the content above.

Reply:
<img width="379" alt="Screenshot on 2021-11-21 at 13 08 00" src="https://user-images.githubusercontent.com/14863373/142775644-8ad0ef46-ea97-4622-8cc5-f3a44a015d47.png">
<img width="420" alt="Screenshot on 2021-11-21 at 13 08 26" src="https://user-images.githubusercontent.com/14863373/142775652-aca35df1-ed5e-4daa-83ef-72cc98595840.png">

IMO the `hr` looks better. I'll definitely add some bottom margin though

Reply:
<img width="439" alt="Screenshot on 2021-11-21 at 13 17 19" src="https://user-images.githubusercontent.com/14863373/142775891-470d630e-1aa6-4bb6-9230-9b2eb0ef258d.png">
Thoughts on this approach?

Reply:
I think that's a good solution and I also like the use of the dotted breaks.